### ISSUE TITLE
Allow TypedDict mixins

### ISSUE THREAD

**Issue by qmmp123** [microsoft/pyright#3978]
I get error `All Base classes for TypedDict classes must also be TypedDict classes`. In my project I need to use typeddict and class with fields

For example:
```python
from typing import TypedDict

class Fields:
  field1: str
  field2: str
  field3: str
  field4: str
  field5: str
  field6: str

class MyTypedDict(TypedDict, Fields):  # Here I get type checking error
  pass

class MyClass(Fields):
  def __init__(self, fields_dict: MyTypedDict):
    fields_dict[' # also here I want to get completions for keys. Now it doesn't working
    for attr, value in fields_dict.items():
      setattr(self, attr, value)
```

**Expected behavior**
Pyright should should give completions and pass type checking

**VS Code extension or command-line**
I use command-line.
Version: 1.1.271
Python: 3.8

**Comment by erictraut** [comment]
This isn't allowed by `TypedDict`. Python enforces that a class cannot inherit from a TypedDict and a non-TypedDict base class. If you run the above code, you will receive a runtime `TypeError`.

**Comment by erictraut** [comment]
The following is allowed by both pyright and the runtime. Perhaps this is what you intend?

```python
from typing import TypedDict

class Fields(TypedDict):
  field1: str
  field2: str
  field3: str
  field4: str
  field5: str
  field6: str

class MyTypedDict(Fields):
  pass
```

**Comment by qmmp123** [comment]
> This isn't allowed by `TypedDict`. Python enforces that a class cannot inherit from a TypedDict and a non-TypedDict base class. If you run the above code, you will receive a runtime `TypeError`.

Try run this code by yourself. You will not get any error.



> The following is allowed by both pyright and the runtime. Perhaps this is what you intend?
> 
> ```python
> from typing import TypedDict
> 
> class Fields(TypedDict):
>   field1: str
>   field2: str
>   field3: str
>   field4: str
>   field5: str
>   field6: str
> 
> class MyTypedDict(Fields):
>   pass
> ```

In your example you missed class `MyClass`

**Comment by qmmp123** [comment]
![image](https://user-images.githubusercontent.com/6360702/191789672-079b73ea-7087-4a83-9f6d-8868bbf50c32.png)

**Comment by erictraut** [comment]
A TypedDict is a structural type that overlays a `dict`. It wouldn't make sense for it to inherit from a non-TypedDict class.

It's possible that the Python implementation you're using permits this, but I'd consider that a bug in the Python implementation. The CPython implementation of TypedDict is correct here. The PEP is quite clear on this matter and the semantics don't make sense otherwise.

**Comment by qmmp123** [comment]
Wow. In Python3.8 you will not get error. This is strange. Ok. Thank you for your quick reply
